Resume 6-12 (#7)
eliannyscer/Top-100-Python-Interview: Resume-6-12.md

# Resume

## Q6. Memory managed

> * The computer memory is like a empty book. Different writers want to write some story there but to do it they need to ask for permission to the book's manager because this show where they can write without over write the other story.

Reviewer: ```suggestion
* The computer memory is like a empty book. Different writers want to write a story, but to do it they need to ask for permission to the book's manager because only he can show where they can write without over write the other story.
```

> * When anybody is reference or using the story somebody delete it to have more space for the new ones.

Reviewer: ```suggestion
* When anybody is referencing or using the story, someone else deletes it to have more space for the new ones.
```

> * `The writer` are the `applications`, `the book's manager` is the `memory manager` and the person that `remove` the story is the `garbage collection`.

Reviewer: ```suggestion
* `The writers` are the `applications`, `the book's manager` is the `memory manager` and the person who `remove` the story is the `garbage collection`.
```

> * Additionally you can think in the `reference` like the `number of the page` in the book where you can find the story(address).

Reviewer: ```suggestion
* Additionally you can think that the `reference` is like the `page number` in the book where you can find start location of the specific story.
```

## Q7. Name-scapes

> This make `immune our programs from name conflicts` because is checking if the names in our program are unique. The name is a unique identifier and the space is the place where you can access to the variable.

Reviewer: ```suggestion
Name-scapes makes `our programs immune from name conflicts` because it is checking if the names in our program are unique. The name is a unique identifier and the space is the place where you can access to the variable.
```

> `Name-scapes is like surname`,if you ask for some student in a classroom probably will be  difficult to find but is you ask for him with name and surname will be easier. The python interpreted understand what exact variable we are trying to find in the code.

Reviewer: ```suggestion
`Name-scapes are like surname`, if you will look for some student only by surname in a classroom probably will be difficult to find, but if you search using with name and surname will be easier. The python interpreted understand exact variables we are trying to find in the code.
```

## Q8. PYTHON-PATH

> Allow users to import modules that have not been installed yet, you can configure to add additional directories where python will look for modules and packages. For most installations, you should not configure these variables because Python knows where to find its standard library.

Reviewer: ```suggestion
Allows users to import modules that have not been installed yet, you can configure to add additional directories where python will look for modules and packages. For most installations you should not configure these variables because Python knows where to find its standard library.
```

## Q9. Python modules

> Is a `.py` file containing executable code or a set of functions you want to include in your application.

Reviewer: ```suggestion
It is file with `.py` extension and contains executable code or a set of functions you want to include in your application.
```

Some of the commonly used modules are:

> `os` -for interacting with operating system.
> 
> `sys` - To manipulate python runtime environment.
> 
> `math` - It is used in mathematical task.
> 
> `random` - This produce a random choice.
> 
> `data time` - To work with data and time data.
> 
> `JSON` - It is used to work with JSON data.

Reviewer: ```suggestion
- `os` - For interacting with operating system.
- `sys` - To manipulate python runtime environment.
- `math` - It is used in mathematical task.
- `random` - For producing a random choice.
- `data time` - For work with time data.
- `JSON` - It is used to work with JSON data.
```

## Q10. Local and global variables

`Local` - Variables declared inside of the function.

> `Global` - Variables declared outside of the function.

Reviewer: Maybe some example?

> ## Q11. Python case sensitive

Reviewer: Maybe some example?

> Yes.Example, if a variable is named ‘HelloWorld‘, then an error will occur if the variable is called ‘helloworld‘. Variables, functions and objects in Python must be called exactly how they are named.

Reviewer: ```suggestion
Yes. For example if you have a variable named ‘HelloWorld‘, an error will occur if you try declare a new variable with the name ‘helloworld‘. Variables, functions and objects in Python must be called exactly how they are named.
```
